plugins/auth-backend/src/providers/saml/provider.ts

```diff
@@ -129,12 +129,13 @@ export const createSamlProvider: AuthProviderFactory = ({
   const providerId = 'saml';
   const entryPoint = config.getString('entryPoint');
   const issuer = config.getString('issuer');
+  const origin = config.getString('origin');
```

Comment:
This should already be available through `globalConfig.appUrl`, better grab it from there to avoid the duplicate config

Reply:
we can't use `globalConfig.appUrl`. We need origin to be `localhost:3000` but setting `appUrl` to `localhost` in docker will not allow external traffic.

Reply:
The `globalConfig.appUrl` is supposed to point to an externally reachable URL. By default it's based on the `app.baseUrl`, which also should be the externally reachable URL. See https://github.com/backstage/backstage/blob/ad88b14b593c2243905ae2de4b8a0705ba781c32/plugins/auth-backend/src/lib/oauth/OAuthAdapter.ts#L54 for how it's set up for OAuth.

Reply:
I'm assuming that's `app.baseUrl`? Are you running the local development `app:serve` in docker in that case? In a production deployment the `app.baseUrl` should have no impact impact on what host the server is bound to.

Assuming the issue is that you're running in docker for local development, I'd perhaps be up for adding `app.listen` with `app.listen.host` and `app.listen.port`, just like `backend.listen`. That config would be picked up by the `app:serve` task in the CLI to override the local development serve host and port.

Reply:
ok

Reply:
@feiming i am trying to setup saml auth via okta, would you happen to have a sample app-config and references I could take a peek at?